engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageState.java

```diff
@@ -180,6 +221,18 @@ public void put(final long key, final MessageRecord record) {
     deadline.wrapLong(record.getDeadline());
     deadlineColumnFamily.insert(deadlineMessageKey, DbNil.INSTANCE);
 
+    long count = 0L;
+    if (!messagesDeadlineCountColumnFamily.isEmpty()) {
+      count = messagesDeadlineCountColumnFamily.get(messagesDeadlineCountKey).getValue();
+      messagesDeadlineCount.wrapLong(count + 1L);
+      messagesDeadlineCountColumnFamily.update(messagesDeadlineCountKey, messagesDeadlineCount);
+    } else {
```

Comment:
:x: We can keep an in-memory `count` and keep it updated, instead of `get` every time. Then no need to check if the key already exists. `get` can be expensive it has to read data that is not in memory.